### PR TITLE
Added two developer guide and removed from README ' builder docker image' and 'launch service from source'

### PR DESCRIPTION
**Pull request by writinwaters**
### Type of change

- [x] Documentation Update

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -302,4 +227,4 @@ See the [RAGFlow Roadmap 2024](https://github.com/infiniflow/ragflow/issues/162)
 
 ## 🙌 Contributing
 
-RAGFlow flourishes via open-source collaboration. In this spirit, we embrace diverse contributions from the community. If you would like to be a part, review our [Contribution Guidelines](./docs/references/CONTRIBUTING.md) first.
+RAGFlow flourishes via open-source collaboration. In this spirit, we embrace diverse contributions from the community. If you would like to be a part, review our [Contribution Guidelines](./docs/develop/CONTRIBUTING.md) first.
```
Comment (JinHai-CN) [inline README.md]: home dir.

File: README_ja.md [hunk 2]
```diff
@@ -286,4 +213,4 @@ $ bash ./entrypoint.sh
 
 ## 🙌 コントリビュート
 
-RAGFlow はオープンソースのコラボレーションによって発展してきました。この精神に基づき、私たちはコミュニティからの多様なコントリビュートを受け入れています。 参加を希望される方は、まず [コントリビューションガイド](./docs/references/CONTRIBUTING.md)をご覧ください。
+RAGFlow はオープンソースのコラボレーションによって発展してきました。この精神に基づき、私たちはコミュニティからの多様なコントリビュートを受け入れています。 参加を希望される方は、まず [コントリビューションガイド](./docs/develop/CONTRIBUTING.md)をご覧ください。
```
Comment (JinHai-CN) [inline README_ja.md]: home.

File: README_ko.md [hunk 3]
```diff
@@ -322,4 +217,4 @@ $ docker compose up -d
 
 ## 🙌 컨트리뷰션
 
-RAGFlow는 오픈소스 협업을 통해 발전합니다. 이러한 정신을 바탕으로, 우리는 커뮤니티의 다양한 기여를 환영합니다. 참여하고 싶으시다면, 먼저 [가이드라인](./docs/references/CONTRIBUTING.md)을 검토해 주세요.
+RAGFlow는 오픈소스 협업을 통해 발전합니다. 이러한 정신을 바탕으로, 우리는 커뮤니티의 다양한 기여를 환영합니다. 참여하고 싶으시다면, 먼저 [가이드라인](./docs/develop/CONTRIBUTING.md)을 검토해 주세요.
```
Comment (JinHai-CN) [inline README_ko.md]: home

File: README_zh.md [hunk 4]
```diff
@@ -313,7 +211,7 @@ $ systemctl start nginx
 
 ## 🙌 贡献指南
 
-RAGFlow 只有通过开源协作才能蓬勃发展。秉持这一精神,我们欢迎来自社区的各种贡献。如果您有意参与其中,请查阅我们的 [贡献者指南](./docs/references/CONTRIBUTING.md) 。
```
Comment (JinHai-CN) [inline README_zh.md:316]: home

File: docs/developer/build_docker_image.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+---
+sidebar_position: 1
+slug: /build_docker_image
+---
+
+# Build a RAGFlow Docker Image
+
+A guide explaining how to build a RAGFlow Docker image from its source code. By following this guide, you'll be equipped to create a local Docker image that can be used for development, debugging, or testing purposes.
```
Comment (JinHai-CN) [inline docs/developer/build_docker_image.md]: equipped => able

File: docs/developer/launch_ragflow_from_source.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
+---
+sidebar_position: 2
+slug: /launch_ragflow_from_source
+---
+
+# Launch the RAGFlow Service from Source 
+
+A guide explaining how to set up a RAGFlow service from its source code. By following this guide, you'll be equipped to debug using the source code.
```
Comment (JinHai-CN) [inline docs/developer/launch_ragflow_from_source.md]: equipped => able

File: docs/developer/launch_ragflow_from_source.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+:::tip NOTE
+If you have not installed Docker on your local machine (Windows, Mac, or Linux), see the [Install Docker Engine](https://docs.docker.com/engine/install/) guide.
+:::
+
+## Launch the Service from Source
+
+To launch the RAGFlow service from source code:
+
+### 1. Clone the RAGFlow Repository
```
Comment (JinHai-CN) [inline docs/developer/launch_ragflow_from_source.md]: Remove number

File: docs/developer/launch_ragflow_from_source.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+   ```
+
+2. Activate Python virtual env
+
+   ```bash
+   source .venv/bin/activate
+   export PYTHONPATH=$(pwd)
+   ```
+
+3.  [Optional] If you are inside GFW
```
Comment (JinHai-CN) [inline docs/developer/launch_ragflow_from_source.md]: If you can't access HuggingFace

File: docs/developer/launch_ragflow_from_source.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+1. Navigate to the web directory:
+
+   ```bash
+   cd web
+   ```
+
+2. Install frontend dependencies:
+
+   ```bash
+   npm install --force
```
Comment (JinHai-CN) [inline docs/developer/launch_ragflow_from_source.md]: Merge with above command

File: docs/developer/launch_ragflow_from_source.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
+   Update proxy.target to http://127.0.0.1:9380
+
+4. Start up the frontend service:
+
+   ```bash
+   npm run dev 
+   ```
+
+### 7. Access the RAGFlow service
```
Comment (JinHai-CN) [inline docs/developer/launch_ragflow_from_source.md:132]: RAGFlow server

File: README.md (hunk 1 above)
Comment (yuzhichang) [reply README.md]: `docs/developer/CONTRIBUTING.md`?

File: docs/developer/build_docker_image.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+:::tip NOTE
+If you have not installed Docker on your local machine (Windows, Mac, or Linux), see the [Install Docker Engine](https://docs.docker.com/engine/install/) guide.
+:::
+
+## Build a RAGFlow Docker Image
+
+To build a RAGFlow Docker image from source code:
+
+### 1. Git Clone the Repository
```
Comment (yuzhichang) [inline docs/developer/build_docker_image.md]: Remove number?

File: docs/developer/launch_ragflow_from_source.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+```bash
+git clone https://github.com/infiniflow/ragflow.git
+cd ragflow/
+```
+
+### 2. Install Python dependencies
+
+1. Install Poetry:
+   
+   ```bash
```
Comment (yuzhichang) [inline docs/developer/launch_ragflow_from_source.md]: Code blocks are not indented uniformly.

File: docs/developer/launch_ragflow_from_source.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+1. Add the following line to `/etc/hosts` to resolve all hosts in **docker/service_conf.yaml** to `127.0.0.1`:
+
+   ```
+   127.0.0.1       es01 mysql minio redis
+   ```
+
+2. In **docker/service_conf.yaml**, update mysql port to `5455` and es port to `1200`, as specified in **docker/.env**.
+
+3. Replace **conf/service_conf.yaml** with the contents of **docker/service_conf.yaml**.
```
Comment (yuzhichang) [inline docs/developer/launch_ragflow_from_source.md]: Remove this item. `conf/service_conf.yaml` is a soft link to `docker/service_conf.yaml`.

**Review by yuzhichang — CHANGES_REQUESTED** [review]

File: docs/developer/build_docker_image.md (hunk 11 above)
Comment (writinwaters) [reply docs/developer/build_docker_image.md]: CONTRIBUTING.md will be moved to the root directory of the project.

**Pull request merged**